aws-crt-http/HttpConnection.cs

```diff
@@ -142,10 +140,49 @@ public struct HttpHeader
         [MarshalAs(UnmanagedType.LPStr)]
         public string Value;
 
+        private Int32 nameSize;
+
+        private Int32 valueSize;

```

Comment:
since `Name` and `Value` are public, can't users change them, and then `nameSize` and `valueSize` will get out of sync?

do we really need these values on the user-facing class?

if we need them, maybe we want a setter to be sure they stay in sync

Reply:
we do need it to stay in sync with native side. added headers